packages/@react-spectrum/utils/src/useResizeObserver.ts

```diff
@@ -0,0 +1,40 @@
+import { RefObject, useEffect } from "react";
+
+function hasResizeObserver() {
+  return typeof window.ResizeObserver !== 'undefined';
+}
+
+type useResizeObserverOptionsType<T> = {
+  ref: RefObject<T>,
+  onResize: () => void
+}
+
+export function useResizeObserver<T extends HTMLElement>(options: useResizeObserverOptionsType<T>) {
+  const {ref, onResize} = options;
+
+  useEffect(() => {
+    if (!ref) { return; }
+
+    if (!hasResizeObserver()) {
+      window.addEventListener('resize', onResize, false);
+      return () => {
+        window.removeEventListener('resize', onResize, false);
+      };
+    } else {
+
+      const resizeObserverInstance = new ResizeObserver((entries) => {
```

Comment:
i think this should be window.ResizeObserver, or both should be without. But it's strange for it to be two ways